pkg/api/apis/operators/installplan_types.go

```diff
@@ -248,6 +251,29 @@ func (b *BundleLookup) SetCondition(cond BundleLookupCondition) BundleLookupCond
 	return cond
 }
 
+func OrderSteps(steps []*Step) []*Step {
+	// CRDs must be applied first
+	crdList := []*Step{}
+
+	// CSVs must be applied second
+	csvList := []*Step{}
+
+	// Other items may be applied in any order
+	remainingItems := []*Step{}
+	for _, step := range steps {
+		switch step.Resource.Kind {
+		case crdKind:
+			crdList = append(crdList, step)
+		case ClusterServiceVersionKind:
+			csvList = append(csvList, step)
+		default:
+			remainingItems = append(remainingItems, step)
+		}
+	}
+	result := append(crdList, csvList...)
```

Comment:
Since the capacity is known up front, it would be better to allocate it and just assign the values in the right order, instead of `append`s (which will copy and expand the slice)

Reply:
Sounds good - will do!